src/main/java/net/ioixd/client/mixin/PokemonUpdateTick.java

```diff
@@ -30,15 +30,71 @@ private void tickMovement(Vec3d movement, CallbackInfo info) {
             Pokemon pokemonData = pokemonEntity.getPokemon();
             Entity firstPassenger = pokemonEntity.getFirstPassenger();
             if (firstPassenger instanceof PlayerEntity player) {
+                
                 // Make the Pokemon's position match the player's and set their position
                 // accordingly.
-                float speedModifier = pokemonData.isLegendary() ? 0.0f
-                        : (float) CobblemountsClient.SYNCED_CONFIG.legendaryModifier;
-                float movementSpeed = player.getMovementSpeed() * (pokemonData.getSpeed() / 12.0f) + speedModifier;
-                if (CobblemountsClient.SYNCED_CONFIG.cappedSpeed) {
-                    if (movementSpeed >= CobblemountsClient.SYNCED_CONFIG.speedCap) {
-                        movementSpeed = (float) CobblemountsClient.SYNCED_CONFIG.speedCap;
+                float legendaryModifier = pokemonData.isLegendary() ? 0.0f
+                    : (float) CobblemountsClient.SYNCED_CONFIG.legendaryModifier;
+                boolean isLegendary = pokemonData.isLegendary();
+                //float movementSpeed = player.getMovementSpeed() * (pokemonData.getSpeed() / 12.0f) + speedModifier;
+                float speedScalar = (float) CobblemountsClient.SYNCED_CONFIG.groundSpeedScalar;
+                float speedCap = (float) CobblemountsClient.SYNCED_CONFIG.groundSpeedCap;
+                boolean isSpeedCapped = CobblemountsClient.SYNCED_CONFIG.groundCappedSpeed;
+                boolean useLogScaling = CobblemountsClient.SYNCED_CONFIG.groundUseLogScaling;
+                float movementSpeed = 0.0f;
+                if (!useLogScaling) {
```

Comment:
The only thing that is different between these two blocks, as far as I can see, is movementSpeed. Please update your code so that only movementSpeed is set and the rest of the code is outside of the block.